src/typeChecker.ts

```diff
@@ -1,49 +1,55 @@
-import fs from 'fs';
 import path from 'path';
 import glob from 'fast-glob';
 import ts from 'typescript';
+import * as ttp from 'typescript-to-proptypes';
 
-let config: ts.CompilerOptions;
-let program: ts.Program;
+const TS_CONFIG = ttp.loadConfig(path.join(process.cwd(), 'tsconfig.json'));
 
-export function loadTSConfig(): ts.CompilerOptions {
-  if (config) {
-    return config;
-  }
-
-  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-  const { config: maybeConfig, error } = ts.readConfigFile(
-    path.join(process.cwd(), 'tsconfig.json'),
-    (filePath) => fs.readFileSync(filePath, 'utf8'),
-  );
-
-  if (error) {
-    throw error;
-  }
-
-  const { options, errors } = ts.parseJsonConfigFileContent(maybeConfig, ts.sys, process.cwd());
-
-  if (errors.length > 0) {
-    throw errors[0];
-  }
-
-  config = options;
+const PROGRAM_LOOKUP: Record<string, ts.Program> = {};
 
-  return options;
+function getProgramCacheKey(pattern: string | string[]) {
+  return typeof pattern === 'string' ? pattern : pattern.join('//');
 }
 
-export function loadProgram(pattern: true | string, root: string): ts.Program {
-  if (program) {
-    return program;
+export function loadProgram(
+  pattern: true | string | string[],
+  root: string,
+): ts.Program {
+  const globSource = pattern === true ? './src/**/*.ts' : pattern;
+  const key = getProgramCacheKey(globSource);
+
+  if (key in PROGRAM_LOOKUP) {
+    return PROGRAM_LOOKUP[key];
   }
 
-  program = ts.createProgram(
-    glob.sync(pattern === true ? './src/**/*.ts' : pattern, {
+  PROGRAM_LOOKUP[key] = ttp.createProgram(
```

Comment:
since the program is built from the glob pattern and the plugin will only be loaded in once for the entire of a given babel build, I figured it'd be safest if we had one program per pattern in the off-chance someone ran this w/ two different patterns